frame/contracts/src/exec.rs

```diff
@@ -1461,6 +1490,31 @@ where
 			current
 		}
 	}
+	fn add_dependency(&mut self, code_hash: CodeHash<Self::T>) -> Result<(), DispatchError> {
+		let frame = self.top_frame_mut();
+		let info = frame.contract_info.get(&frame.account_id);
+		let owner_info = OwnerInfoOf::<T>::get(code_hash).ok_or(Error::<T>::ContractNotFound)?;
+		let deposit = Perbill::from_percent(30).mul_ceil(owner_info.deposit());
```

Comment:
is there a good place to define this 30% constant? Or should we make it part of the config?